.github/workflows/ci.yml

```diff
@@ -177,30 +177,60 @@ jobs:
 
   java-unit-tests:
     if: needs.detect-changes.outputs.java-code-changes == 'true'
-    needs: [detect-changes]
-    runs-on: gcp-perf-core-16-default
-    timeout-minutes: 30
+    name: "[UT] ${{ matrix.name }}"
+    needs: [ detect-changes ]
+    timeout-minutes: 25
     permissions: {}  # GITHUB_TOKEN unused in this job
+    runs-on: ${{ matrix.runs-on }}
+    strategy:
+      fail-fast: false
+      matrix:
+        group: [ general, zeebe, client ]
+        include:
+          - group: general
+            name: "General Unit Tests"
+            # Here we exclude all modules that are separated in extra jobs below
+            # Example: '[-+]<groupId>:artifactId' -> '-' excludes
+            maven-modules: "'-:zeebe-workflow-engine','-:zeebe-broker','-:camunda-process-test-java','-:zeebe-gateway-rest','-:camunda-client-java','-:zeebe-elasticsearch-exporter','-:zeebe-opensearch-exporter','-:camunda-exporter','-:zeebe-atomix-cluster','-:zeebe-cluster-config'"
```

Comment:
I did a longer research, but didn't found a better way to exclude modules, except for having extra maven profiles, etc. Which I think is also sub-optimal, so I went this approach.